oozie-to-airflow/mappers/base_mapper.py

```diff
@@ -69,6 +69,20 @@ def last_task_id(self) -> str:
         """
         return self.name
 
+    def on_parse_node(self, workflow):
+        """
+        Called when processing a node.
+
+        This is a good time to read all data from a node.
```

Comment:
I think that all data should be read from the node in this place. This will allow for easier separation of roles between each method. This will also simplify the tests because it will not be necessary to build complex XML structures. Tests will be able to be written more easily.

Reply:
Ok, maybe in the future yes. But for now we don't do this. So this comment is misleading, because it doesn't reflect the current state of affairs.

Reply:
In my opinion, it is not misleading. It describes our goal.  When the next person joins the team he will know that he should use this method to implement the logic. 

I agree that old mappers do not use this method, but use old mechanisms. We are in the transition stage. We realized that using one method we are not able to do everything. We must expand the functionality of our converter in order to be able to implement all functionalities.

@potiuk What do you think?

Reply:
When the next person joins the team he will look at it, and be confused:
"Wait... it says we should read *all* data from a node here, however in all of the use cases that I can find in the code only a small part of node data is read... 🤔 "

When we change the logic to really read *all* data here in all use cases, *then* we can have such a comment.